mmv1/third_party/terraform/tests/resource_google_service_account_test.go

```diff
@@ -84,6 +84,65 @@ func TestAccServiceAccount_basic(t *testing.T) {
 	})
 }
 
+func TestAccServiceAccount_Disabled(t *testing.T) {
+	t.Parallel()
+
+	accountId := "a" + randString(t, 10)
+	uniqueId := ""
+	displayName := "Terraform Test"
+	desc := "test description"
+	project := getTestProjectFromEnv()
+	expectedEmail := fmt.Sprintf("%s@%s.iam.gserviceaccount.com", accountId, project)
+	vcrTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			// The first step creates a basic service account
+			{
+				Config: testAccServiceAccountBasic(accountId, displayName, desc),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"google_service_account.acceptance", "project", project),
+				),
+			},
+			{
+				ResourceName:      "google_service_account.acceptance",
+				ImportStateId:     fmt.Sprintf("projects/%s/serviceAccounts/%s", project, expectedEmail),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// The second step disables the service account
+			{
+				Config: testAccServiceAccountDisabled(accountId, displayName, desc, true),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"google_service_account.acceptance", "project", project),
+					testAccStoreServiceAccountUniqueId(&uniqueId),
+				),
+			},
+			{
+				ResourceName:      "google_service_account.acceptance",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// The third step enables the disabled service account
+			{
+				Config: testAccServiceAccountDisabled(accountId, displayName, desc, false),
```

Comment:
Should this go back to testAccServiceAccountBasic?

Reply:
It has the same effect. Disabled is set to false by default.

Reply:
Ah, I was confused because the name of the function is `testAccServiceAccountDisabled` so I assumed it would cause the service account to be disabled. What would you think about changing the name to `testAccServiceAccountWithDisabled` or similar? Otherwise I'm fine merging with the current name.

Reply:
Lets merge it as it is.